index.js

```diff
@@ -1,93 +1,60 @@
-const fs = require("fs"),
-    http = require("http"),
-    path = require("path"),
-    methods = require("methods"),
-    express = require("express"),
-    bodyParser = require("body-parser"),
-    session = require("express-session"),
-    cors = require("cors"),
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
+import express from 'express';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+import morgan from 'morgan';
+import requestId from 'express-request-id';
+import routes from './src/routes/index';
+import Response from './src/helpers/response';
 
-const isProduction = process.env.NODE_ENV === "production";
-
-// Create global app object
 const app = express();
 
-app.use(cors());
-
-// Normal express config defaults
-app.use(require("morgan")("dev"));
-app.use(bodyParser.urlencoded({ extended: false }));
-app.use(bodyParser.json());
-
-app.use(require("method-override")());
-app.use(express.static(__dirname + "/public"));
-
-app.use(
-    session({
-        secret: "authorshaven",
-        cookie: { maxAge: 60000 },
-        resave: false,
-        saveUninitialized: false
-    })
-);
+const port = process.env.PORT || 9000;
 
-if (!isProduction) {
-    app.use(errorhandler());
-}
+let response;
 
-if (isProduction) {
-    mongoose.connect(process.env.MONGODB_URI);
-} else {
-    mongoose.connect("mongodb://localhost/conduit");
-    mongoose.set("debug", true);
-}
-
-require("./models/User");
+app.use(cors());
 
-app.use(require("./routes"));
+app.use(requestId());
 
-/// catch 404 and forward to error handler
-app.use(function(req, res, next) {
-    const err = new Error("Not Found");
-    err.status = 404;
-    next(err);
-});
+morgan.token('id', req => req.id);
 
-/// error handlers
+const loggerFormat = ':id [:date[web]] ":method :url" :status :response-time';
 
-// development error handler
-// will print stacktrace
-if (!isProduction) {
-    app.use(function(err, req, res, next) {
-        console.log(err.stack);
+app.use(morgan(loggerFormat, {
+  skip: (req, res) => res.statusCode < 400, stream: process.stderr
+}));
 
-        res.status(err.status || 500);
+app.use(morgan(loggerFormat, {
+  skip: (req, res) => res.statusCode >= 400, stream: process.stdout
+}));
 
-        res.json({
-            errors: {
-                message: err.message,
-                error: err
-            }
-        });
-    });
-}
+app.use(bodyParser.urlencoded({ extended: false }));
+app.use(bodyParser.json());
 
-// production error handler
-// no stacktraces leaked to user
-app.use(function(err, req, res, next) {
-    res.status(err.status || 500);
-    res.json({
-        errors: {
-            message: err.message,
-            error: {}
-        }
-    });
+app.use('/api/v1/', routes);
+
+app.all('/', ((req, res) => {
+  response = new Response(
```

Comment:
Why do you need a separate class for this?

Reply:
A response is sent every time a request comes into the server. This response has a defined set of keys (status, code, message, and data). Creating a class prevents duplication of having to define those keys in multiple parts of the application. Instead, an instance of the class is created for reuse. It also gives room for extension by creating multiple behaviors in any case where the response keys need additional keys for another type of response like an ErrorResponse can inherit from its parent Response.

Reply:
Sweet!